src/main/java/com/esri/samples/symbology/read_symbols_from_mobile_style_file/ReadSymbolsFromMobileStyleFileController.java

```diff
@@ -0,0 +1,353 @@
+        }
+      });
+    });
+  }
+
+  /**
+   * Gets the keys of the selected symbol components, and constructs a multilayer point symbol.
+   */
+  @FXML
+  private void buildCompositeSymbol() {
+
+    // retrieve the requested keys for the selected hat, eyes and mouth symbols
+    String hatKey = getRequestedSymbolKey(hatSelectionListView);
+    String eyesKey = getRequestedSymbolKey(eyesSelectionListView);
+    String mouthKey = getRequestedSymbolKey(mouthSelectionListView);
+
+    // create a list of keys to use for getting the requested symbol.
+    List<String> symbolKeys = Arrays.asList("Face1", eyesKey, mouthKey, hatKey);
+
+    // get the symbol from the SymbolStyle
+    ListenableFuture<Symbol> symbolFuture = emojiStyle.getSymbolAsync(symbolKeys);
+    symbolFuture.addDoneListener(() -> {
+      try {
+        faceSymbol = (MultilayerPointSymbol) symbolFuture.get();
+        if (faceSymbol == null) {
+          return;
+        }
+
+        // loop through all the symbol layers and lock the color
+        faceSymbol.getSymbolLayers().forEach(symbolLayer -> symbolLayer.setColorLocked(true));
+
+        // unlock the color of the base layer. Changing the symbol layer will now only change this layer's color
+        faceSymbol.getSymbolLayers().get(0).setColorLocked(false);
+
+        // set the color of the symbol
+        faceSymbol.setColor(colorSelectionListView.getSelectionModel().getSelectedItem());
+
+        // set the size of the symbol
+        faceSymbol.setSize((float) sizeSlider.getValue());
+
+        // update the symbol preview
+        updateSymbolPreview(faceSymbol);
+
+      } catch (ExecutionException | InterruptedException e) {
+        new Alert(Alert.AlertType.ERROR, "Error creating symbol with the provided symbol keys" + e.getMessage()).show();
+      }
+    });
+  }
+
+  /**
+   * Returns the symbol key of the symbol style search result that is currently selected in the list view.
+   * @param listView the list view of which to query the selected item
+   * @return String representing the key of the selected symbol style
+   */
+  private String getRequestedSymbolKey(ListView<SymbolStyleSearchResult> listView) {
+    SymbolStyleSearchResult requestedSymbol = listView.getSelectionModel().getSelectedItem();
+    return (requestedSymbol != null ? requestedSymbol.getKey() : "");
+  }
+
+  /**
+   * Creates an ImageView object from a provided symbol and displays it in the preview area.
+   *
+   * @param multilayerPointSymbol the symbol used to create the image view
+   */
+  private void updateSymbolPreview(MultilayerPointSymbol multilayerPointSymbol) {
+    // remove the previously displayed image view
+    symbolPreviewHBox.getChildren().clear();
+
+    // create an image and image view from the symbol
+    ListenableFuture<Image> symbolImageFuture = multilayerPointSymbol.createSwatchAsync(0x00000000, 1);
+    try {
+      Image symbolImage = symbolImageFuture.get();
```

Comment:
Since you're not using a DoneListener, the `createSwatchAsync` will run on the main thread. Is that deliberate? Probably best to promote the pattern of using DoneListeners for all async operations?

Reply:
Possibly just an oversight on my part, or in the sample design. Added the `doneListener`, as you suggested.

Reply:
On testing this, it seems that running the `createSwatchAsync` on the main thread might have been deliberate. Putting the block in a `doneListener` now makes the symbol preview flicker and show graphical bugs. I'll do some invesitgating on why.